File: gardenstore/src/app/models/dto/produit.dto.ts
export interface ProduitDTO {
    id: number;
    nom: string;
    reference?: number;
    stock: number;
    prixDeVente: number;
    avis: string;
    categorie: ProduitCategorie;
}

export enum ProduitCategorie {
    OUTILLAGE = "OUTILLAGE",
    MOBILIER = "MOBILIER",
    PLANTES = "PLANTES",
    ACCESSOIRES = "ACCESSOIRES"
}

File: proj.gardenstore/src/main/java/gs/java/fs/api/models/dtos/ProduitDTO.java
package gs.java.fs.api.models.dtos;

import gs.java.fs.domain.entities.Description;
import gs.java.fs.domain.entities.Produit;


public record ProduitDTO(Long id,
                         String nom,

                         Integer reference,
                         Integer prixDeVente,

                         Integer stock,

                         String avis,

                         Description categorie

                         )
{
    public static ProduitDTO fromEntity(Produit produit) {
        if (produit == null)
            return null;

        return new ProduitDTO(
                produit.getId(),
                produit.getNom(),
                produit.getReference(),
                produit.getPrixDeVente(),
                produit.getStock(),
                produit.getAvis(),
                produit.getCategorie()
                );
    }

}